pkg/auth/providers/migration/activedirectory_migration.go

```diff
@@ -66,11 +66,13 @@ func MigrateActiveDirectoryDNToGUID(ctx context.Context, management *config.Mana
 			backOffDuration := time.Second * 30
 			var err error
 			for steps > 0 {
+				logrus.Infof("Active directory principalID migration starting")
 				err = m.migrate()
 				if err != nil {
 					time.Sleep(backOffDuration)
 					backOffDuration = 2 * backOffDuration
 				} else {
+					logrus.Infof("Active directory principalID migration complete")
```

Comment:
This looks like it would report migration complete 5 times, once per each step. Was that intended? Seems like we maybe shouldn't call it complete until the whole thing is finished?

Reply:
That block is only hit if migrate() completes without error, so we will only see it once.

Reply:
the steps only come into play if we errored out and try it again.